OMCompiler/Compiler/runtime/systemimpl.c

```diff
@@ -261,23 +261,80 @@ extern int SystemImpl__setLDFlags(const char *str)
   return 0;
 }
 
+#if defined(__MINGW32__) || defined(_MSC_VER)
+extern int SystemImpl__stringToUnicodeSize(const char* str)
+{
+  return MultiByteToWideChar(CP_UTF8, 0, str, -1, NULL, 0);
+}
+
+extern int SystemImpl__stringToUnicode(const char* str, wchar_t* unicode, int size)
+{
+  return MultiByteToWideChar(CP_UTF8, 0, str, -1, unicode, size);
+}
+
+int SystemImpl__unicodeToStringSize(const wchar_t* unicode)
+{
+  return WideCharToMultiByte(CP_UTF8, 0, unicode, -1, NULL, 0, NULL, NULL);
+}
+
+int SystemImpl__unicodeToString(const wchar_t* unicode, char* str, int size)
+{
+  return WideCharToMultiByte(CP_UTF8, 0, unicode, -1, str, size, NULL, NULL);
+}
+
+/* Make sure windows paths use frontslash and not backslash */
+void SystemImpl__toWindowsSeperators(char* buffer, int bufferLength)
+{
+  int i;
+  for (i=0; i<bufferLength && buffer[i]; i++) {
+    if (buffer[i] == '\\') buffer[i] = '/';
+  }
+}
+#endif
+
+int SystemImpl__chdir(const char* path)
+{
+#if defined(__MINGW32__) || defined(_MSC_VER)
+  int unicodePathLength = SystemImpl__stringToUnicodeSize(path);
+  wchar_t unicodePath[unicodePathLength];
+  SystemImpl__stringToUnicode(path, unicodePath, unicodePathLength);
+
+  if (!SetCurrentDirectoryW(unicodePath)) {
+    fprintf(stderr, "SetCurrentDirectoryW failed. %lu\n", GetLastError());
+    return -1;
+  }
+  return 0;
+#else
+  if (!chdir(path)) {
+    fprintf(stderr, "chdir failed.\n");
```

Comment:
Don't use stderr for error message. Use the Error.mo or ignore them. In this case chdir returns true or false so omc should know what to do.